src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -77,8 +78,12 @@ public async Task<IEnumerable<Type>> GetExtensionsStartupTypesAsync()
                     {
                         bindingsSet.Add(binding.Type);
                     }
+                    isPrecompiledFunctionApp = isPrecompiledFunctionApp || functionMetadata.Language == DotNetScriptTypes.DotNetAssembly;
                 }
+            }
 
+            if (bundleConfigured && (!isPrecompiledFunctionApp || _extensionBundleManager.IsLegacyExtensionBundle()))
```

Comment:
So if it's precompiled and a legacy bundle, we'll still load bundles? What scenario does that cover? Are people doing that?

Reply:
Yeah, it is not something bundles was designed for. However there are a lot of used doing that right now.  If we stop loading bundles for them it would be a breaking change.